components/AppBar.js

```diff
@@ -121,64 +166,86 @@ export default class Appbar extends React.Component {
    * Operates on an instance of a state variable and plots in accordance
    */
   clickAction ($instance$) {
-    // TODO handle time series loading with redux actions
-    this.plotInstance($instance$);
+    const { 
+      widgets, createWidget, requestDataRetrieve,
+      currentSelectedPlotInstancePath, 
+      changeInstancePathOfCurrentSelectedPlot,
+    } = this.props;
+
+    const instancePath = $instance$.getInstancePath();
+
+    const widget = widgets.find(({ id }) => id == instancePath);
+
+    if (currentSelectedPlotInstancePath != instancePath) {
+      changeInstancePathOfCurrentSelectedPlot(instancePath)
+    }
+    
+    if (widget){
+      if (widget.status == 'DESTROYED') {
+        const [ , , tsName ] = instancePath.split('.', );
+        createWidget({
+          name: tsName,
+          instancePath,
+          id: instancePath,
+        })
+      }
+    } else {
+      requestDataRetrieve()
+    }
+
     this.refs.controlpanelref.close();
   }
 
   plotInstance ($instance$) {
```

Comment:
I'd like to move back business logic important stuff from the appBar. It's not intuitive to find and does not seem to be right place. I'm still wondering what is better though: following the discussion about redux-react best practices I'll see now what I can find to move that into a react independent controller system, that listens to the current state and take actions in response.

Reply:
there is a corner case that forced me to put that logic... Imagine that the user goes to the controlPanel and clicks timeseries1 icon. Then a new panel with the widget opens up and the user decides to close it. Then he goes again to the controlPanel and clicks on the same plot...

In that case, from the redux state point of view, if I just update the currentSelectedPlot state with the new value, I will get the same state than before, therefore I will not be able to open up the plot again (prevProps and props are the same)

There are other ways to come up with a different set of states that will address that corner case, but other corner cases will follow. Maybe we allocate some time to reflect on what the ideal redux set of states will look like, so that we make sure no corner cases are left behind

Reply:
Agree. We can think about using a middleware on order to avoid these "in the middle" states that have the duty to remember the last action to do something else.